components/molecules/HighlightInput/highlight-input-form.tsx

```diff
@@ -642,7 +642,9 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
                 </Tooltip>
                 <TextInput
                   id="highlight-link-input"
-                  className={`text-sm shadow-none h-10 flex-none ${!isHighlightURLValid && "border-red-500"}`}
+                  state={isHighlightURLValid ? "valid" : "invalid"}
+                  className={`text-sm shadow-none h-10 flex-none ${errorMsg && "border-red-500"}`}
+                  // className="text-sm shadow-none h-10 flex-none"
```

Comment:
refactor (blocking): `errorMsg && "border-red-500` will result in `undefined` for a class name if there is no error message.

```suggestion
                  state={isHighlightURLValid ? "valid" : "invalid"}
                  className={`text-sm shadow-none h-10 flex-none ${errorMsg ? "border-red-500": ""}`}
```

You could also use `clsx` which is already in the project as well.

```suggestion
                  state={isHighlightURLValid ? "valid" : "invalid"}
                  className={clsx("text-sm shadow-none h-10 flex-none", errorMsg ? "border-red-500": "")}
```